show_client/mac_cli.go

```diff
@@ -46,3 +48,113 @@ func getMacAgingTime(options sdc.OptionMap) ([]byte, error) {
 }
+
+// MacEntry represents a single FDB entry
+type MacEntry struct {
+	Vlan       string `json:"vlan"`
+	MacAddress string `json:"macAddress"`
+	Port       string `json:"port"`
+	Type       string `json:"type"`
+}
+
+const (
+	ApplDb   = "APPL_DB"
+	StateDb  = "STATE_DB"
+	FDBTable = "FDB_TABLE"
+)
+
+// getMacTable queries APPL_DB and STATE_DB FDB_TABLE entries and returns either the list or count per options
+func getMacTable(options sdc.OptionMap) ([]byte, error) {
+	// Parse filters
+	vlanFilter := -1
+	if v, ok := options["vlan"].Int(); ok {
+		vlanFilter = v
+	}
+	portFilter, _ := options["port"].String()
+	addrFilter, _ := options["address"].String()
+	typeFilter, _ := options["type"].String()
+	wantCount, _ := options["count"].Bool()
+
+	// Fetch APPL_DB and STATE_DB separately to preserve source and avoid key collisions
+	applData, err := GetMapFromQueries([][]string{{ApplDb, FDBTable}})
+	if err != nil {
+		log.Errorf("Unable to get APPL_DB FDB_TABLE, err: %v", err)
+		return nil, err
+	}
+	stateData, err := GetMapFromQueries([][]string{{StateDb, FDBTable}})
+	if err != nil {
+		log.Errorf("Unable to get STATE_DB FDB_TABLE, err: %v", err)
+		return nil, err
+	}
+
+	// Prefer APPL_DB entries on duplicates; track seen keys "vlan|mac"
+	seen := make(map[string]struct{})
+	entries := make([]MacEntry, 0, len(applData)+len(stateData))
+
+	addIfMatch := func(vlan, macAddress, port, mtype string) {
+		// Filters
+		if vlanFilter >= 0 && vlan != fmt.Sprint(vlanFilter) {
+			return
+		}
+		if portFilter != "" && !strings.EqualFold(port, portFilter) {
+			return
+		}
+		if addrFilter != "" && !strings.EqualFold(strings.ToLower(addrFilter), strings.ToLower(macAddress)) {
+			return
+		}
+		if typeFilter != "" && strings.ToLower(typeFilter) != strings.ToLower(mtype) {
+			return
+		}
+		key := vlan + "|" + strings.ToLower(macAddress)
+		if _, exists := seen[key]; exists {
+			return
+		}
+		seen[key] = struct{}{}
+		entries = append(entries, MacEntry{
+			Vlan:       vlan,
+			MacAddress: macAddress,
+			Port:       port,
+			Type:       strings.ToLower(mtype),
+		})
+	}
+
+	ProcessFDBData(applData, ApplDb, addIfMatch)
+	ProcessFDBData(stateData, StateDb, addIfMatch)
+
+	if wantCount {
+		resp := map[string]int{"count": len(entries)}
+		return json.Marshal(resp)
+	}
+
+	sort.Slice(entries, func(i, j int) bool {
+		if entries[i].Vlan == entries[j].Vlan {
+			return strings.ToLower(entries[i].MacAddress) < strings.ToLower(entries[j].MacAddress)
+		}
+		return entries[i].Vlan < entries[j].Vlan
+	})
+	return json.Marshal(entries)
+}
+
+func ProcessFDBData(data map[string]interface{}, source string, addIfMatch func(string, string, string, string)) {
+	for k, v := range data {
+		fv, ok := v.(map[string]interface{})
+		if !ok {
+			continue
+		}
+		vlan, mac, ok := ParseKey(k)
+		if !ok {
+			continue
+		}
+		addIfMatch(vlan, mac, toString(fv["port"]), toString(fv["type"]))
+	}
+}
+
+func ParseKey(k string) (vlan string, mac string, ok bool) {
```

Comment:
Can you make this a general function where we would need to split keys by delimitor? Dont make it specific to vlan/mac address. Please move to show_common.go after